### PR TITLE
feat: add emotes

### PR DESCRIPTION
**Pull request by pedrotambo**
This PR adds the Emote schema and its versioning. More details: https://docs.google.com/document/d/1OZ2uwYjFKsYXhzZB-zxCLlRe01DZ2yYo-2FeM5TVVZA/edit?usp=sharing 
@decentraland/dapps

### CODE REVIEW

File: src/platform/item/emote/emote.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    type: 'object',
+    properties: {
+      ...baseItemProperties,
+      ...standardProperties,
+      ...thirdPartyProps,
+      emoteDataADR73: EmoteDataADR73.schema
+    },
+    additionalProperties: true,
+    required: [...requiredBaseItemProps],
+    oneOf: [
```
Comment (marianogoldman) [inline src/platform/item/emote/emote.ts:30]: The error reporting of oneOf is not very nice and barely useful. I'd suggest adding `errorMessage` property like the one below.

File: src/platform/item/wearable/wearable.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            minItems: 1
+          },
+          category: WearableCategory.schema
+        },
+        required: ['replaces', 'hides', 'tags', 'representations', 'category']
+      }
+    },
+    additionalProperties: true,
+    required: [...requiredBaseItemProps, 'data'],
+    oneOf: [
```
Comment (marianogoldman) [inline src/platform/item/wearable/wearable.ts:65]: Same here. Something along the lines of "either properties for a standard wearable or properties for third party wearable are required"

File: src/platform/item/wearable/wearable.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        errorMessage:
+          'for third party wearables "collectionAddress" and "rarity" are not allowed'
+      }
+    ]
+  }
+
+  const _isThirdPartyKeywordDef = {
+    keyword: '_isThirdParty',
+    validate: (schema: boolean, data: any) => !schema || isThirdParty(data),
+    errors: false
```
Comment (marianogoldman) [inline src/platform/item/wearable/wearable.ts:85]: Shouldn't this report errors if validation fails?

File: test/platform/item/body-shape.spec.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  const bodyShape: BodyShape = BodyShape.FEMALE
+
+  testTypeSignature(BodyShape, bodyShape)
+
+  it('static tests must pass', () => {
+    expect(BodyShape.validate(bodyShape)).toEqual(true)
+    expect(BodyShape.validate(null)).toEqual(false)
+    expect(BodyShape.validate({})).toEqual(false)
+  })
+})
```
Comment (marianogoldman) [inline test/platform/item/body-shape.spec.ts:15]: I suggest adding assertions for the errors when `validate` returns `false`. This way we can tackle upfront the quality of error messages returned.

File: src/platform/item/emote/emote.ts (hunk 1 above)
Comment (pedrotambo) [reply src/platform/item/emote/emote.ts:30]: 👍

File: test/platform/item/emote/adr-95/representation.spec.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+  testTypeSignature(EmoteRepresentationADR73, representation)
+
+  it('static tests must pass', () => {
+    expect(EmoteRepresentationADR73.validate(representation)).toEqual(true)
+    expect(EmoteRepresentationADR73.validate(null)).toEqual(false)
+    expect(EmoteRepresentationADR73.validate({})).toEqual(false)
+  })
+
+  it('representation without body shape fails', () => {
```
Comment (marianogoldman) [inline test/platform/item/emote/adr-95/representation.spec.ts:23]: Same here, for all the validations returning false, I'd add a check for the array of errors returned.

File: test/platform/item/body-shape.spec.ts (hunk 4 above)
Comment (pedrotambo) [reply test/platform/item/body-shape.spec.ts:15]: 👍

File: src/validation.ts [hunk 6]
```diff
@@ -40,7 +40,7 @@ export function generateValidator<T>(
 ): ValidateFunction<T> {
   const ajv = new Ajv({ $data: true, allErrors: true })
   ajv_keywords(ajv)
-  ajv_errors(ajv)
+  ajv_errors(ajv, { singleError: true })
```
Comment (marianogoldman) [inline src/validation.ts:43]: Not sure what this does exactly, but doesn't it go against the `allErrors: true` that we have 2 lines above? I think it's important that all errors are reported.

Comment (pedrotambo) [reply src/validation.ts:43]: It joins errors with `;` just in very specific cases where the `instancePath` is the same

**Pull request merged**